spring-batch-docs/asciidoc/readersAndWriters.adoc

```diff
@@ -2937,27 +2488,20 @@ public FooService fooService() {
 ----
 
 [[validatingInput]]
-
-
 === Validating Input
 
-During the course of this chapter, multiple approaches to parsing
-input have been discussed. Each major implementation throws an
-exception if it is not 'well-formed'. The
-`FixedLengthTokenizer` throws an exception if a
-range of data is missing. Similarly, attempting to access an index in a
-`RowMapper` of `FieldSetMapper`
-that does not exist or is in a different format than the one expected
-causes an exception to be thrown. All of these types of exceptions are
-thrown before `read` returns. However, they do not
-address the issue of whether or not the returned item is valid. For
-example, if one of the fields is an age, it obviously cannot be negative.
-It may parse correctly, because it exists and is a number, but it does not
-cause an exception. Since there are already a plethora of validation
-frameworks, Spring Batch does not attempt to provide yet another.
-Rather, it provides a simple interface, called `Validator`, that can be implemented by any
-number of frameworks, as shown in the following interface definition:
-
+During the course of this chapter, multiple approaches to parsing input have been
+discussed. Each major implementation throws an exception if it is not 'well-formed'. The
+`FixedLengthTokenizer` throws an exception if a range of data is missing. Similarly,
+attempting to access an index in a `RowMapper` of `FieldSetMapper` that does not exist or
```

Comment:
should be RowMapper `or` instead of  RowMapper `of`

will fix on merge.

Reply:
Yup. Thanks.